cmd/thanos/store.go

```diff
@@ -431,8 +434,10 @@ func runStore(
 	{
 		ins := extpromhttp.NewInstrumentationMiddleware(reg, nil)
 
-		compactorView := ui.NewBucketUI(logger, conf.webConfig.externalPrefix, conf.webConfig.prefixHeaderName, conf.component)
-		compactorView.Register(r, ins)
+		if !conf.disableWeb {
+			compactorView := ui.NewBucketUI(logger, conf.webConfig.externalPrefix, conf.webConfig.prefixHeaderName, conf.component)
+			compactorView.Register(r, ins)
+		}
```

Comment:
So you don't want to disable the api this time?

Reply:
Yes

Reply:
What's the reason for this? We should be consistent with the compactor change.

Reply:
The reason is that disabling the API will not save more resources. 

From a consistency point of view, disabling is also reasonable, I will change it.

Reply:
Done